frontend/src/lib/WidgetStateManager.ts

```diff
@@ -193,6 +194,32 @@ export class WidgetStateManager {
     })
   }
 
+  public sendUploadFileMessage(
+    id: string,
+    name: string,
+    lastModified: number,
+    data: Uint8Array
+  ): void {
+    const limit = 10 * 1e6 //10MB
+
+    const newFileMessage: NewFile = new NewFile()
```

Comment:
nit: `const newFileMessage = new NewFile()` (no need to specify the type twice, since it can be inferred from the constructor)